core/src/main/java/io/questdb/cutlass/text/TextImportTask.java

```diff
@@ -0,0 +1,992 @@
+                        long nextOffset = nextLengthAndOffset & MASK;
+
+                        //line indexing stops on first EOL char, e.g. \r, but it could be followed by \n
+                        long diff = nextOffset - offset - bytesToRead;
+                        if (diff > -1 && diff < 2 && bytesToRead + nextLineLength < fileBufSize) {
+                            bytesToRead += diff + nextLineLength;
+                            additionalLines++;
+                        } else {
+                            break;
+                        }
+                    }
+
+                    i += additionalLines;
+
+                    long n = ff.read(fd, fileBufAddr, bytesToRead, offset);
+                    if (n > 0) {
+                        lexer.parse(fileBufAddr, fileBufAddr + n, Integer.MAX_VALUE, this::onFieldsPartitioned);
+                    } else {
+                        throw TextException.$("Can't read from file path='").put(path).put("', errno=").put(ff.errno()).put(",offset=").put(offset);
+                    }
+                }
+
+                lexer.parseLast();
+            } finally {
+                if (fd > -1) {
+                    ff.close(fd);
+                }
+            }
+        }
+
+        private void mergePartitionIndexAndImportData(
+                final FilesFacade ff,
+                Path partitionPath,
+                final TextLexer lexer,
+                long fileBufAddr,
+                long fileBufSize,
+                DirectCharSink utf8Sink,
+                DirectLongList mergeIndexes,
+                Path tmpPath
+        ) throws TextException {
+            mergeIndexes.clear();
+            partitionPath.slash$();
+            int partitionLen = partitionPath.length();
+
+            long mergedIndexSize = -1;
+            long address = -1;
+            long fd = -1;
+
+            try {
+                mergedIndexSize = openIndexChunks(ff, partitionPath, mergeIndexes, partitionLen);
+
+                partitionPath.trimTo(partitionLen);
+                partitionPath.concat(CsvFileIndexer.INDEX_FILE_NAME).$();
+
+                fd = TableUtils.openFileRWOrFail(ff, partitionPath, CairoConfiguration.O_NONE);
+                address = TableUtils.mapRW(ff, fd, mergedIndexSize, MemoryTag.MMAP_PARALLEL_IMPORT);
+
+                long mergedAddr = getMergedIndexes(ff, address, mergeIndexes);
+
+                importPartitionData(lexer, mergedAddr, mergedIndexSize, fileBufAddr, fileBufSize, utf8Sink, tmpPath);
+            } finally {
+                if (fd > -1) {
+                    ff.close(fd);
+                }
+                if (address != -1) {
+                    ff.munmap(address, mergedIndexSize, MemoryTag.MMAP_PARALLEL_IMPORT);
+                }
+                unmap(ff, mergeIndexes);
+            }
+        }
+
+        private long getMergedIndexes(FilesFacade ff, long address, DirectLongList mergeIndexes) {
```

Comment:
confusing method name, it does not imply that indexing takes place. Perhaps this method should be called "mergeIndexes". I would also make arguments consistent. `address` is the destination memory for sorted data, `mergeIndexes` is the source. This method sig is (to, from) thereas underlying JNI method sig is (from, to). This makes code harder to follow.

Also, lets rename `mergeIndexes` to `unmergedIndexes`

Reply:
fixed